### PR TITLE
Added register classes.

### PR DESCRIPTION
**Pull request by Thordreck**
- Added registers classes (V and I) and tests.
- Added some simple random number generation utilities (there's an instruction that stores a random number in a register).

### CODE REVIEW

File: include/registers/DataRegister.hpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            ~DataRegister() = default;
+
+            void Set(uint8_t _value);
+            void Set(const DataRegister& _register);
+
+            bool Add(uint8_t _value);
+            bool Add(const DataRegister& _register);
+
+            bool Substract(uint8_t _value);
+            bool Substract(const DataRegister& _register);
```
Comment (blasco) [inline include/registers/DataRegister.hpp]: Subtract

**Review by blasco — CHANGES_REQUESTED** [review]
Only that small typo. I'm not fully convince of the advantage of wrapping all of this. The data registers are just a uint8_t, right? why not just use a uint8_t?

**Comment by blasco** [comment]
I think now that I've seen the implementation I get the point of having this class. Basically to manage the carry bits etc. I guess we could add a c++17 [[nodiscard]] to enforce taking care of these carrying bits. 

When I saw that they returned bool I just thought it was some kind of flag to say 'successful' operation, 
 maybe giving a name to that bool could could add expressiveness to the code.

```
using Carry_t = bool;
using Borrow_t = bool;

[[nodiscard]] Carry_t      Add (uint8_t _value) 
[[nodiscard]] Borrow_t   Subtract (uint8_t _value)
```

Just a small suggestion, but now, just looking at the header, one can deduce what the implementation is doing without looking at it.

File: include/registers/DataRegister.hpp (hunk 1 above)
Comment (Thordreck) [reply include/registers/DataRegister.hpp]: Fixed.

**Comment by Thordreck** [comment]
I have fixed the typo you pointed out. Some comments below:
>Basically to manage the carry bits etc. I guess we could add a c++17 [[nodiscard]] to enforce taking care of these carrying bits.

Instruction 7XNN do not take into account if a carry occurs. Adding `[[nodiscard]]` would issue a warning in that case (and with our current compiler flags it would turn into an error).
> giving a name to that bool could could add expressiveness to the code.

I don't really have a strong opinion on the alias you propose. I can add them if you want.

**Review by blasco — APPROVED** [review]

**Pull request merged**